Exception in workflow service REST endpoint /update

**Describe the bug**

The REST endpoint to update a workflow throws a 500 error.
When the workflow instance is parsed from XML, the operations do not contain a reference to the workflow instance so the database update throws an error because the workflow_id field is null.

**To Reproduce**

Steps to reproduce the behavior:
1. Update a workflow via the workflow REST endpoint: update
2. Get a 500 error

**Expected behavior**

The workflow is updated with no errors.

**Stack trace**

```
...
Caused by: org.opencastproject.workflow.api.WorkflowDatabaseException: Could not update workflow with ID '536814'
        at org.opencastproject.workflow.api.WorkflowServiceDatabaseImpl.updateInDatabase(WorkflowServiceDatabaseImpl.java:336)
        at org.opencastproject.workflow.impl.WorkflowServiceImpl.update(WorkflowServiceImpl.java:1280)
        ... 150 more
Caused by: javax.persistence.RollbackException: Exception [EclipseLink-4002] (Eclipse Persistence Services - 2.7.11.v20220804-52dea2a3c0): org.eclipse.persistence.exceptions.DatabaseException
Internal Exception: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Column 'workflow_id' cannot be null
Error Code: 1048
Call: INSERT INTO oc_workflow_operation (id, abortable, continuable, completed, started, description, exception_handler_workflow, if_condition, execution_host, fail_on_error, failed_attempts, job, max_attempts, retry_strategy, state, template, time_in_queue, workflow_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
        bind => [18 parameters bound]
Query: InsertObjectQuery(operation:'include, state:'SUCCEEDED')
        at org.eclipse.persistence.internal.jpa.transaction.EntityTransactionImpl.commit(EntityTransactionImpl.java:161)
        at org.opencastproject.db.DBSessionImpl.execTxChecked(DBSessionImpl.java:153)
        at org.opencastproject.db.DBSessionImpl.execTx(DBSessionImpl.java:126)
        at org.opencastproject.db.DBSessionImpl.execTx(DBSessionImpl.java:99)
        at org.opencastproject.db.DBSessionImpl.execTx(DBSessionImpl.java:89)
        at org.opencastproject.workflow.api.WorkflowServiceDatabaseImpl.updateInDatabase(WorkflowServiceDatabaseImpl.java:327)
        ... 151 more

```

--- a/modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/JaxbWorkflowInstance.java
+++ b/modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/JaxbWorkflowInstance.java
@@ -123,14 +123,18 @@ public class JaxbWorkflowInstance {
   }
 
   public WorkflowInstance toWorkflowInstance() {
-    return new WorkflowInstance(id, state, template, title, description, creatorName, organizationId, dateCreated,
+    WorkflowInstance wf = new WorkflowInstance(id, state, template, title, description, creatorName, organizationId, dateCreated,
             dateCompleted, mediaPackage,
             Optional.ofNullable(operations).orElseGet(Collections::emptyList)
-                    .stream().map(JaxbWorkflowOperationInstance::toWorkflowOperationInstance).collect(Collectors.toList()),
+                    .stream().map(JaxbWorkflowOperationInstance::toWorkflowOperationInstance)
+                    .collect(Collectors.toList()),
             Optional.ofNullable(configurations).orElseGet(Collections::emptySet)
                     .stream()
                     .collect(Collectors.toMap(JaxbWorkflowConfiguration::getKey, JaxbWorkflowConfiguration::getValue)),
             mediaPackageId, seriesId);
+    // Set workflow reference in each operation
+    wf.getOperations().stream().forEach(wo -> wo.setWorkflowInstance(wf));
+    return wf;
   }
 
   @Override

